### ISSUE TITLE
Warlock 2.0

### ISSUE THREAD

**Issue by warius1983** [datchannin/Brothers-WoW#57]
I was rethinking a little the lock a little and how to make it spicier.

Warlock must burn a lot more fatique, he must kill him self and get healed back, actually he has almost infinite fatique for little health, and woth tier he doesn't feel pain.

I think the perk tree can stay like it is, the problem is to rework all the spells.

Shadowbolt: it musn't be a clone of a big fireball, but rather a clone of a  pyroblast, and I mean that it should cost really a ton of fatique (70-75) and some turns of cooldown, you shouldn't use it every round.
Damage will be changed accordingly.

Life Tap: actually it deals 10 hp for 20 fatique, change it, like 30 hp for 15 fatique, we need a worse ratio to make it work, he can always use healing to regain health, to regain maybe proper fatique he should use drain mana

Drain Soul: Ok this spell at the moment is lame, it doesn't scale with spell power and very little maybe with level it should raise by level and maybe perk. My Idea is to change it completely, it has a cost of 9 AP and 20 fat.
Why 9 AP cost? because if you have a warlock 15 ( max level ) and the spell deals 40-50 it must kill the target to get the soul shard, if you kill the enemy with the cast, you regain some AP ( 4 ) like berserking, and can cast some empowered spell.
For example enemy low health , start of the turn 9 AP spent in one cast of drain soul, enemy dies, you gain soul shard and 4 AP, what do you do? an empowered curse of agony? move your warlock ? it is your choice. Like it is now it's weak and bad but the design behind the soul shard is perfect. ( master demonolgy perk!!! for next turn use).

Drain Life and Mana: they too need some more concrete scaling, and with the help of drain funnel.
Thinking about numbers drain life should cost only AP and no fatique, and should be the base damaging spell of the lock for single target, the one that can be cast each round regardless any cooldowns. Drain mana should cost more AP and regain some more fatique.

Now curse of agony: I tested with 3 AP per cast and set 4 turns, this can be changed of course, it is the unique spell of the warlock, ad the idea of the build is to dot a lot of enemies when for example he can't use shadowbolt. 3AP 15 fat base.

Then some perks should be used to raise the damage/turn but also increasing the cost in fatique, so the locks needs to life tap more to get back more fatique, big powers come in cost of great sacrifices in blood.

Then soulshard could increase the power of the curse of agony by 100%, 50% of shadowbolt, and maybe 200% of drain life and mana?

What do you say do you like this philosophy? If you want i can write some numbers with the scaling the spells already have

**Comment by datchannin** [comment]
ok.
What we have now.
For Warlock should be done two specs.
Now there are:
1. Destruction (shadowbolt, crit chance, crit value)
2. Afflication (curse of agony, soul shard as power, curse of doom)
Warlock has number of perks for fatigue restore - life tap and mana drain
And number of perks for health restore - health drain and demon skin

The problems are.
1. Shadowbolt damage is fully random, so it is more profitable to have second mage, it's damage is predictable 
-> destruction spec is useless

2. Curse of agony has the same damage as shadowbolt, but it starts damage dealing on the next turn and duration is several turns.
Curse - (14-18)x2 turns = 28-36
Shadowbolt - 20-35

Compare them.
First turn. Shadowbolt damage ~ 27, Agony ~ 0 (put on first target)
Second turn. Shadowbolt ~ 54, Agony ~ 16 (and put on second target)
Third turn. Shadowbolt ~ 81, Agony ~ 32 (put on third target)
Forth turn. Shadowbolt ~ 108, Agony ~ 64
...
May be damage became equal after turn 6 or more. But combat duration can be 3-4 turns.

I've created soul shard for highly increase curse damage. But it is difficult to get shard (you said that Soul Drain damage is low).
-> afflication spec is useless

3. Now mana tap is the best for fatigue restore and fatigue spend speed is low. 
(-> mana drain is useless)

4. Demon Skin can regain HP only if it was casted with Soul Shard
-> demon skin is useless

then.
What I want:
Create two independence specs (can be the same as now - Destruction and Afflication)
All of these specs should use a lot of fatigue and player should regain fatigue vs hitpoints
As consequence, player should regain hitpoints with some spells or add one more healer for Warlock in group.

Lets discuss Destruction.
I agree that shadowbolt should not be fireball copy. That's why I've added Crit chance.
Now shadowbolt casted 1 per turn (and can crit), fireball casted 3 per turn (with less damage without crit)

Fatigue cost can be increased. But in this case Warlock will not be able to cast Shadowbolt unless he is not 7+ level (get some perks and fatigue points)
May be shadowbolt damage should grow with warlock level and fatigue cost should grow with warlock level.
So than more level you are, than more fatigue you spend, but you will be able to restore it.
Cooldown can be added, but not 2 turns like pyroblast. May be 1.

Or for help warlock cast shadowbolt on low level, it can starts with 3AP cost and cost should be increased with every new Shadowbolt Perk.
So from the start he will cast 3 shadowbolt, like mage's frostbolt
After he obtain crit chance and AP cost will be increased +1 AP, only 2 per turn without CD
Than he obtain more power and more crit, +1 AP, +1AP, only 1 per turn without CD
And at the end Ruin perk, triple crit damage, or even more, but 1 turn CD will be added, and +1 AP cost (7-8 cost at the end, CD per turn and tonnes fatigue that should be restored on next turn)

Life Tap. Numbers should be reworked.
Drain Mana and Drain Life. I don't know. These spells should regain HP and MP for warlock, but Drain Life damages targets, so this is damage spell.
May be these spells should be used only on friendly targets?

Demon Skin. Should be used without soul shard. This is way to regain some HP over turn.
2 turns, 5/5 melee/ranged defense, 5 hp regain for Demon Armor
4 turns, 15/15 melee/ranged defense, 7 hp regain for Improved Demon Skin
and some bonuses from tier may be
Something like this.

Curse of Agony. Can be updated for 2/3 casts per turn with the same damage as now.
May be 4 AP cost from the start (2 per turn), -1 AP cost from additional perk (3 per turn), or may be 5 from start (1 per turn), -1 from perk (2 per turn), -1 from another perk (3 per turn)
Turns duration also can be increased. 2 turns from start, +1 turn from additional perk.
This gives warlock a chance to play with Curse of Agony, without Shadowbolt at all. And deal almost AOE-type damage.

Curse of Doom, should be removed may be. Instead of these, I'll add AOE Curse of Agony, like Frost Nova. 6 targets from 1 cast. And consume Soul Shard.

What about drain soul - it can be as you described. Like more damage and scale it from warlock level, consume soul shard if killed. And spend more AP for use it.
Now damage is 10-12, 4 per turn = 40-48, spend 15 per cast = 60 fatigue
Can be 20-25, 4 AP cost, two times per turn = 40-50 damage.
So it will be (5-10 + warlock level) - ((5-10)+15 level) = 20-25 per cast.

And regain 2 AP if you kill target.
1 cast and kill -> you spend 4AP, regain 2AP. Have 7 AP. Empowered Agony(3) + Normal Agony(3).
2 casts and kill -> you spend 8AP, regain 2AP. Have 3 AP. Empowered Agony (3).

And soul shard should not affected Shadowbolt. This is two different specs.
Warlock can be fully destruction with Crit Chance and Crit Power, and learn base Curse of Agony (5 AP, weak damage, without shards)
Or Warlock can be fully afflication with Multi-Dot-Agony, Massive AOE Agony + Shard, Empowered Agony (Shard), and learn base Shadowbolt (average damage, without crit)

**Comment by warius1983** [comment]
For destruction Shadowbolt:

_"Fatigue cost can be increased. But in this case Warlock will not be able to cast Shadowbolt unless he is not 7+ level (get some perks and fatigue points)
May be shadowbolt damage should grow with warlock level and fatigue cost should grow with warlock level.
So than more level you are, than more fatigue you spend, but you will be able to restore it.
Cooldown can be added, but not 2 turns like pyroblast. May be 1."_

I agree on this one, cooldown and growing fatique/damage/level is the best solution + perk bonus, morphing small shadowbolts into one very big the more you play can be confusing.


_"Life Tap. Numbers should be reworked.
Drain Mana and Drain Life. I don't know. These spells should regain HP and MP for warlock, but Drain Life damages targets, so this is damage spell.
May be these spells should be used only on friendly targets?"_

You showed me a different vision, Drain mana could be an inverse innervate, the lock for example restore 70 fatique, but friendly target loses 35 fatique, could work but has to be much more effective than innervate of course, warrior could be a nice battery for warlock, same for drain life.

_"Curse of Agony. Can be updated for 2/3 casts per turn with the same damage as now.
May be 4 AP cost from the start (2 per turn), -1 AP cost from additional perk (3 per turn), or may be 5 from start (1 per turn), -1 from perk (2 per turn), -1 from another perk (3 per turn)
Turns duration also can be increased. 2 turns from start, +1 turn from additional perk.
This gives warlock a chance to play with Curse of Agony, without Shadowbolt at all. And deal almost AOE-type damage."_

In my opinion curse of doom has saved me many times against gnoll brutes, it's an exceptional skill if well used, I'd save it, The most important thing to consider  is to have all perks active to activate for example aoe curse of agony ( SEED OF CORRUPTION is the spell in WOW ) can use the structure of circle of healing or frost Nova, and maybe need the use of the soulstone.

In conclusion soulstone should be needed for:

empower next shadowbolt by 50%
empower next single curse of agony by 100% ?
cast curse of doom
Cast SEED of CORRUPTION

_What about drain soul - it can be as you described. Like more damage and scale it from warlock level, consume soul shard if killed. And spend more AP for use it._

Not very clear what you meant.

Anyway on tree perk now we should have in first perk tier curse of agony and drain soul
second tier shadowbolt and drainlife/mana
7th tier ( last ) seed of destruction, curse of doom, ruin?

**Comment by warius1983** [comment]
Ok the version has been released ad ready to commented:

Now the perk tree is much more and much more cleaner than before, Great Job.

Ok let's see the spells:

Shadowbolt build:

Assuming shadowbolt has 0% chance to crit as base, and i like the way it grows in faqtique while levelling, this is a summary of the shadowbolt perk:


shadowbolt------->devastation--------->shadowburn--------->Shadow Mastery---------> shadowfury---------->Ruin-------->tot
3AP                              1AP                                                                      1AP                                 1AP
                                  25%crit ch                                                       25% crit ch                     10% crit ch                  
                               100% cr  dam                                                                                                                          100% cr dam
                                                                 % 25Damage                                                             5 base dam
                                                                                                                                                                                    1turn cd

Now after some tests with a levelled warlock I see the spread between minimum damage  and maximal damage is HUGE, and the DPS seems to be a little incosistent, to be frank, the only thing that matters is does it crit?                                        


How I'd like it:

shadowbolt------->devastation--------->shadowburn--------->Shadow Mastery---------> shadowfury---------->Ruin
6AP
                                 30% crit ch                                                      20% crit ch                       10% crit ch
                                                                                                                                                                                 100 crit dam
                                +x base damage      +4x base dam           +2x base damage            +3x base damage
1 turn cd

                                                                        
xbase damage maybe can be 5? just tossing numbers, we need it, stronger base damage but not so uber high crits.



Curse of Agony:

Id' change base range to 4
Fel domination: 8 spell damage +5 fat., and a small istant damage for every cast on the target
Unstable affliction: change + 5 fat.
Seed of Corruption: perfect as it is for now

Life tap: perfect with mana feed

Drain life/mana Drain funnel
These spells actually do not scale with spell power, and they should fight against life tap, so if they do not scale, I'd change change drain funnel to lower AP by 1 for those 2 spells, I beleive they may be not to efficient, I will try


You made a very good revamp of the class in very little time let me know how your tests are going

**Comment by warius1983** [comment]
shadowbolt has no interaction with sol shard right now right? maybe link it in ruin?

**Comment by datchannin** [comment]
Curse of Agony has 3 base range + 3 range from perk + 2 range from enchant = 8. This is enough 9 is too big value.
Shadowbolt should not use Soul Shard, this is for Affliction spec.
May be one more perk should be added, or some updated for increase Curse of Agony duration from 2 to 3.
